scripts/config-tuning/select_best_config.py

```diff
@@ -51,7 +52,14 @@ def guess_gcn_architecture(path: str, raw_json_data: dict):
     gb_per_s_series.name = 'gb_per_s'
 
     # Extract the groups of the regex match to a DataFrame
-    name_regex = r'^(?P<generator>\S+?)_(?P<distribution>uniform|normal|log_normal|poisson)_(?P<value_type>(?>unsigned_)?(?>int|short|char|long_long|float|half|double))_t(?P<block_size>\d+)_b(?P<grid_size>\d+)'
+    
+    # need at least python 3.11 to have possesive quantifier and atomic grouping
+    if version_info.major >= 3 and version_info.minor >= 11:
+        name_regex = r'^(?P<generator>\S+?)_(?P<distribution>uniform|normal|log_normal|poisson)_(?P<value_type>(?>unsigned_)?(?>int|short|char|long_long|float|half|double))_t(?P<block_size>\d+)_b(?P<grid_size>\d+)'
+    else:
+        name_regex = r'^(?P<generator>\S+?)_(?P<distribution>uniform|normal|log_normal|poisson)_(?P<value_type>(unsigned_)?(int|short|char|long_long|float|half|double))_t(?P<block_size>\d+)_b(?P<grid_size>\d+)'
```

Comment:
Might as well just remove the atomic grouping completely if the latter works:

```suggestion
    name_regex = r'^(?P<generator>\S+?)_(?P<distribution>uniform|normal|log_normal|poisson)_(?P<value_type>(unsigned_)?(int|short|char|long_long|float|half|double))_t(?P<block_size>\d+)_b(?P<grid_size>\d+)'
```

Reply:
Removed atomic grouping as suggested in latest commit

Reply:
Thanks!